File: Day064_2025/Q064_GFG.java
import java.util.*;

class Q064_GFG {
    public int longestStringChain(String[] words) {
        Arrays.sort(words, Comparator.comparingInt(String::length));
        Map<String, Integer> dp = new HashMap<>();
        int maxLength = 1;
        
        for (String word : words) {
            int length = 1;
            for (int i = 0; i < word.length(); i++) {
                String sub = word.substring(0, i) + word.substring(i + 1);
                length = Math.max(length, dp.getOrDefault(sub, 0) + 1);
            }
            dp.put(word, length);
            maxLength = Math.max(maxLength, length);
        }
        return maxLength;
    }
}

File: Day064_2025/Q064_GFG.py
class Solution:
    def longestStringChain(self, words):
        words.sort(key=len)
        dp = {}
        max_length = 1

        for word in words:
            length = 1
            for i in range(len(word)):
                sub = word[:i] + word[i+1:]
                length = max(length, dp.get(sub, 0) + 1)
            dp[word] = length
            max_length = max(max_length, length)
        
        return max_length
